### PR TITLE
Move `preact-render-to-string` to `preact/server`

### PR DESCRIPTION
**Pull request by marvinhagemeister**
This PR merges the https://github.com/preactjs/preact-render-to-string repo into core under `preact/server`. All commits from that repo are preserved and all tags have been renamed to `server-x.x.x`.

On top of that there are a few commits that integrate the tooling into our main one. It has not been updated for internals yet, so I've skipped a few tests. The tests can be run via `npm run test:mocha:server`

### CODE REVIEW

File: server/README.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+# preact-render-to-string
```
Comment (JoviDeCroock) [inline server/README.md:1]: We should probably redo this `preact/server` and maybe something with the logos

File: server/package.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{
+	"name": "preact-server",
+	"amdName": "preactServer",
+	"version": "5.1.20",
```
Comment (JoviDeCroock) [inline server/package.json:4]: are we publishing this separately or are we going to keep this at its own version

Comment (marvinhagemeister) [reply server/package.json:4]: I think we can safely ignore this as the plan is to publish this as part of core and not as a separate package. The other addons have bogus version numbers to which don't match the same package. So far we haven't gotten an issue for that so I'm assuming that no tool reads it.

File: server/README.md (hunk 1 above)
Comment (marvinhagemeister) [reply server/README.md]: Good point, I've updated the title and removed the badges. In the longterm we might want to think about removing it or adding readme's for the other addons too. But that's something for later.

**Review by JoviDeCroock — APPROVED** [review]

**Pull request merged**